mne/io/eeglab/eeglab.py

```diff
@@ -0,0 +1,360 @@
+        dict(auditory=1, visual=3). If int, a dict will be created with
+        the id as string. If a list, all events with the IDs specified
+        in the list are used. If None, the event_id is constructed from the
+        EEGLAB (.set) file with each descriptions copied from `eventtype`.
+    montage : str | None | instance of montage
+        Path or instance of montage containing electrode positions.
+        If None, sensor locations are (0,0,0). See the documentation of
+        :func:`mne.channels.read_montage` for more information.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    epochs : instance of Epochs
+        The epochs.
+
+    Notes
+    -----
+    .. versionadded:: 0.11.0
+
+
+    See Also
+    --------
+    mne.Epochs : Documentation of attribute and methods.
+    """
+    epochs = EpochsEEGLAB(input_fname=input_fname, events=events,
+                          event_id=event_id, montage=montage, verbose=verbose)
+    return epochs
+
+
+class RawEEGLAB(_BaseRaw):
+    """Raw object from EEGLAB .set file.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to the .set file.
+    montage : str | None | instance of montage
+        Path or instance of montage containing electrode positions.
+        If None, sensor locations are (0,0,0). See the documentation of
+        :func:`mne.channels.read_montage` for more information.
+    eog : list or tuple
+        Names of channels or list of indices that should be designated
+        EOG channels. If None (default), the channel names beginning with
+        ``EOG`` are used.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of RawEEGLAB
+        A Raw object containing EEGLAB .set data.
+
+    Notes
+    -----
+    .. versionadded:: 0.11.0
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    @verbose
+    def __init__(self, input_fname, montage, eog=None, preload=False,
+                 verbose=None):
+        """Read EEGLAB .set file.
+        """
+        from scipy import io
+        basedir = op.dirname(input_fname)
+        eeg = io.loadmat(input_fname, struct_as_record=False,
+                         squeeze_me=True)['EEG']
+        if eeg.trials != 1:
+            raise TypeError('The number of trials is %d. It must be 1 for raw'
+                            ' files' % eeg.trials)
```

Comment:
Does this mean it's likely an `epochs` file? If so, you could suggest the user try using `read_epochs_eeglab` instead.